shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/DynamicFeatureManager.java

```diff
@@ -0,0 +1,128 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.dynamicfeatures;
+
+
+/**
+ * Basic interface that handles downloading and loading of dynamic features.
+ *
+ * <p>Flutter dynamic feature support is still in early developer preview and should
+ * not be used in production apps yet.
+ *
+ * <p>The Flutter default implementation is PlayStoreDynamicFeatureManager.
+ *
+ * <p>DynamicFeatureManager handles the embedder/android level tasks of downloading, installing, and
+ * loading dart deferred libraries. A typical code-flow begins with a dart call to loadLibrary() on
+ * deferred imported library. See https://dart.dev/guides/language/language-tour#deferred-loading
+ * This call retrieves a unique identifier called the loading unit id, which is assigned by
+ * gen_snapshot during compilation. The loading unit id is passed down through the engine and invokes
+ * donwloadDynamicFeature. Once the feature module is downloaded, loadAssets and loadDartLibrary are
+ * invoked which discovers, unpacks, and opens any assets or shared libs included in the dynamic
+ * feature module and passes the results back to the engine.
+ *
+ * <p>The methods here may also be called manually via platform channel messages.
+ *
+ * <p>A dynamic feature module is uniquely identified by a module name. Each feature module may
+ * contain one or more loading units, uniquely identified by the loading unit ID.
+ */
+public interface DynamicFeatureManager {
+  /**
+   * Request that the feature module be downloaded and installed.
+   *
+   * <p>This method is called by Flutter when loadLibrary() is called on a dart deferred library
+   * which relies on this method to download the requested module under the hood. This method may
+   * also be manually invoked via a system channel message.
+   *
+   * <p>This method begins the download and installation of the specified feature module. For example,
+   * the Play Store dynamic delivery implementation uses SplitInstallManager to request the download of
+   * the module. Download is not complete when this method returns. Upon completion of
+   * download, loadAssets and loadDartLibrary should be called to complete the dynamic feature load
```

Comment:
called by whom? The subclass implementation or the engine?

Reply:
Also it's a bit confusing because it sounds like you're asking the implementers to call loadAssets and loadDartLibrary. But they're implementing loadAssets and loadDartLibrary :D. So what should those implementations then call? If they need to then transitively call something else, might as well state it here and explain why those are broken out.